### DISCUSSION TITLE
How to set handleCli to true?

### DISCUSSION THREAD

**Discussion by NiklasBr** [pimcore/pimcore#17597] [Q&A / Support]
From the logs when saving/updating an object:

```log
Not saving object_3241691 to cache as process is running in CLI mode (pass force to override or set handleCli to true)
```

But we cannot find the configuration option to _actually_ set `handleCli` to `true`, how is it intended to work?

**Comment by NiklasBr** [comment]
The only way I found has been the below example in the `Kernel` class, but [that relies on an internal method](https://github.com/pimcore/pimcore/blob/8b1dd697b2fa5c8ffee83946e251298fbce3a3ec/lib/Cache.php#L32-L45):

```php
    public function boot(): void
    {
        parent::boot();
        if (PHP_SAPI === "cli") {
            Cache::getHandler()->setHandleCli(true);
        }
    }
```

What other ways are there which does not rely on `@internal`-tagged functionality?

**Comment by GAL-AL** [comment]
Could this issue be related?

https://github.com/pimcore/pimcore/issues/17639#issuecomment-2370345928

**Reply by NiklasBr** [reply]
Not for my question it seems. The log message you see in my first post here is expected and mostly works as intended, I just want to enable (Redis) cache in the CLI because a lot of stuff is being processed by CLI scripts and they need the cache to perform well.
